src/main/java/fi/aalto/cs/intellij/activities/RequiredPluginsCheckerActivity.java

```diff
@@ -0,0 +1,213 @@
+package fi.aalto.cs.intellij.activities;
+
+import static java.util.stream.Collectors.toMap;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.ide.plugins.PluginManagerConfigurable;
+import com.intellij.ide.plugins.RepositoryHelper;
+import com.intellij.ide.plugins.newui.BgProgressIndicator;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationAction;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupActivity;
+import com.intellij.openapi.updateSettings.impl.PluginDownloader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.StringJoiner;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A startup activity for the plugin.
+ */
+public class RequiredPluginsCheckerActivity implements StartupActivity {
+
+  private static final Logger logger = LoggerFactory
+      .getLogger(RequiredPluginsCheckerActivity.class);
+
+  private final Map<String, String> requiredPluginNames = new HashMap<>();
+  private Map<String, String> missingOrDisabledPluginNames = new HashMap<>();
+  private List<IdeaPluginDescriptor> missingOrDisabledIdeaPluginDescriptors;
+  private List<IdeaPluginDescriptor> availableIdeaPluginDescriptors;
+
+  /**
+   * An actual startup work gets done here.
+   */
+  @Override
+  public void runActivity(@NotNull Project project) {
```

Comment:
I'd consider structuring this so that you do not need the class attributes. E.g. the method populateRequiredPluginsMap would be colled getRequiredPluginsMap (and return the current set). Then, you would pass this to the getMissingOrDisabledPlugins method that'd return a list of missing or disabled plugins based on the parameter (merging methods filterMissingOrDisabledPluginNames and createListOfMissingOrDisabledPluginDescriptors). Then, finally, this list would be passed to the checkPluginsStatusAndNotify-method, which would iterate the list and act accordingly.